test/e2e/network/netpol/policies.go

```diff
@@ -18,564 +18,259 @@ package netpol
 
 import (
 	"fmt"
-
 	"k8s.io/apimachinery/pkg/util/intstr"
 
 	v1 "k8s.io/api/core/v1"
 	networkingv1 "k8s.io/api/networking/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type SetFunc func(policy *networkingv1.NetworkPolicy)
+
+func CreateNetworkPolicy(fn ...SetFunc) *networkingv1.NetworkPolicy {
+	policy := &networkingv1.NetworkPolicy{}
+	for _, f := range fn {
+		f(policy)
+	}
+	return policy
+}
+
+func SetObjectMetaName(name string) SetFunc {
+	return func(policy *networkingv1.NetworkPolicy) {
+		policy.ObjectMeta.Name = name
+	}
+}
+
+func SetSpecPodSelector(targetSelector metav1.LabelSelector) SetFunc {
+	return func(policy *networkingv1.NetworkPolicy) {
+		policy.Spec.PodSelector = targetSelector
+	}
+}
+
+func SetSpecPodSelectorMatchLabels(targetLabels map[string]string) SetFunc {
```

Comment:
Same for SpecPodSelectorMatchLabels, the only dynamic part are the network policy rules